apps/keyboard/js/layouts/th.js

```diff
@@ -0,0 +1,134 @@
+Keyboards.th = {
+  label: 'Thai',
+  shortLabel: 'Th',
+  menuLabel: 'ไทย', /*ไทย*/
+  secondLayout: true,
+  basicLayoutKey: 'กขค', /*กขค*/
+  imEngine: 'latin',
+  types: ['text', 'url', 'email'],
+  width: 11,
+  keys: [
+    [
+      { value: 'ๆ' }, { value: 'ภ' }, { value: 'ถ' }, { value: 'ุ' },
+      { value: 'ู' } , { value: 'ึ' }, { value: 'ค' } , { value: 'ต' },
+      { value: 'จ' }, { value: 'ข' }, { value: 'ช' }
+    ], [
+      { value: 'ไ' }, { value: 'ำ' }, { value: 'พ' }, { value: 'ะ' },
+      { value: 'ั' } , { value: 'ี' }, { value: 'ร' }, { value: 'น' },
+      { value: 'ย' }, { value: 'บ' }, { value: 'ล' }
+    ], [
+      { value: 'ฟ' }, { value: 'ห' }, { value: 'ก' }, { value: 'ด' },
+      { value: 'เ' } , { value: '้' }, { value: '่' }, { value: 'า' },
+      { value: 'ส' }, { value: 'ว' }, { value: 'ง' }
+    ], [
+      { value: '⇪', keyCode: KeyEvent.DOM_VK_CAPS_LOCK },
+      { value: 'ป' }, { value: 'แ' }, { value: 'อ' }, { value: 'ิ' },
+      { value: 'ื' }, { value: 'ท' }, { value: 'ม' },
+      { value: 'ใ' }, { value: 'ฝ' },
+      { value: '⌫', keyCode: KeyEvent.DOM_VK_BACK_SPACE }
+    ], [
+      { value: '&nbsp', ratio: 9, keyCode: KeyboardEvent.DOM_VK_SPACE },
+      { value: '↵', ratio: 2, keyCode: KeyEvent.DOM_VK_RETURN }
+    ]
+  ],
+  upperCase: {
+    'ๆ': '฿',
+    'ภ': '1',
```

Comment:
We would have numbers in the symbolLayout (or page[1]), so would not recommend we repeat them here, but I know nothing about Thai, so feel free to keep it if you think this is good enough.